roles/network_plugin/weave/tasks/seed.yml

```diff
@@ -0,0 +1,50 @@
+---
+- name: Weave seed | Set seed if first time
+  set_fact:
+    seed: '{% for host in groups["k8s-cluster"] %}{{ hostvars[host]["ansible_default_ipv4"]["macaddress"] }}{% if not loop.last %},{% endif %}{% endfor %}'
+  when: "weave_seed == 'unset'"
+  run_once: true
+  tags: confweave
+
+- name: Weave seed | Set seed if not first time
+  set_fact:
+    seed: '{{ weave_seed }}'
+  when: "weave_seed != 'unset'"
+  run_once: true
+  tags: confweave
+
+- name: Weave seed | Set peers if fist time
+  set_fact:
+    peers: '{{ weave_ip_current_cluster }}'
+  when: "weave_peers == 'unset'"
+  run_once: true
+  tags: confweave
+
+- name: Weave seed | Set peers if existing peers
+  set_fact:
+    peers: '{{ weave_peers }}{% if weave_ip_current_cluster not in weave_peers %} {{ weave_ip_current_cluster }}{% endif %}'
+  when: "weave_peers != 'unset'"
+  run_once: true
+  tags: confweave
+
+- name: Weave seed | Save seed
+  lineinfile:
+    dest: "./inventory/group_vars/k8s-cluster.yml"
+    state: present
+    regexp: '^weave_seed:'
+    line: 'weave_seed: {{ seed }}'
+  become_user: $USER
```

Comment:
I'm not sure how much I like this approach, but just set `become: no` is easier.